### PR TITLE
Tslint

### PR DESCRIPTION
**Pull request by wzalazar**
This PR introduces

- Configuration Tslint with the same rules that has Frost web
- Report about  issues fixed 
- Report about issues without fix

**239 errors without fix**

103 expected call-signature: '*' to have a typedef
23 Block contains only one statement; remove the curly braces.
21 Imported upper-level module into non-root directory using a relative path
17 Modifying properties of existing object not allowed.
10 expected call-signature to have a typedef
9 Don't use 'Object' as a type. Avoid using the `Object` type. Did you mean `object`?
8 Method signature is mutable, use property signature with readonly modifier instead.
8 Imported module into root directory using a relative path. Please use an absolute path
6 Missing radix parameter
5 Use of default exports is forbidden
5 '*' is declared but its value is never read.
3 'State' is declared but its value is never read.
3 unused expression, expected an assignment or function call
3 A maximum of 1 class per file is allowed.
2 Declaration of public instance method not allowed after declaration of private instance method
2 block is empty
2 Declaration of public static field not allowed after declarationof private instance field.
2 Reassigning parameter 'targetLength' is forbidden.
2 comment must start with a space
2 Reassigning parameter 'padString' is forbidden.
1 Exceeds maximum line length of 120
1 Forbidden bitwise operation
1 == should be ===

**Automatically fixed**

Fixed 63 error(s) in actions/index.ts
Fixed 49 error(s) in components/organisms/WorksByProfile.tsx
Fixed 40 error(s) in components/ResourceProvider.tsx
Fixed 35 error(s) in extensions/Array.ts
Fixed 34 error(s) in components/organisms/Navbar.tsx
Fixed 33 error(s) in components/atoms/CopyableText.tsx
Fixed 31 error(s) in pages/Works/Works.tsx
Fixed 31 error(s) in store.ts
Fixed 31 error(s) in pages/Work/Overview.tsx
Fixed 29 error(s) in components/molecules/Pagination.tsx
Fixed 29 error(s) in reducers/FetchReducer.ts
Fixed 27 error(s) in sagas/FetchSaga.ts
Fixed 22 error(s) in components/atoms/Work.tsx
Fixed 18 error(s) in pages/Works/Layout.tsx
Fixed 17 error(s) in components/molecules/LatestWorks.tsx
Fixed 16 error(s) in pages/Works/Filters.tsx
Fixed 15 error(s) in pages/Work/Tabs/ContentTab.tsx
Fixed 15 error(s) in pages/Work/Tabs/TechnicalTab.tsx
Fixed 14 error(s) in index.tsx
Fixed 14 error(s) in helpers/PoetApi.ts
Fixed 13 error(s) in components/atoms/DatePickerInput.tsx
Fixed 12 error(s) in pages/Works/Loader.tsx
Fixed 12 error(s) in sagas/CacheInvalidationSaga.ts
Fixed 11 error(s) in components/PageLoader.tsx
Fixed 11 error(s) in pages/Landing/LandingLoggedIn.tsx
Fixed 10 error(s) in sagas/NavbarSaga.tsFixed 21 error(s) in extensions/String.ts
Fixed 10 error(s) in images/Images.tsx
Fixed 10 error(s) in pages/Work/Layout.tsx
Fixed 8 error(s) in helpers/UrlObject.ts
Fixed 8 error(s) in extensions/Map.ts
Fixed 8 error(s) in components/Root.tsx
Fixed 7 error(s) in store/PoetAppState.ts
Fixed 7 error(s) in components/organisms/Footer.tsx
Fixed 7 error(s) in pages/Landing/Loader.tsx
Fixed 6 error(s) in pages/Work/Loader.tsx
Fixed 5 error(s) in components/atoms/base/PoetApiResource.ts
Fixed 4 error(s) in components/atoms/Claim.tsx
Fixed 4 error(s) in typings/require.ts
Fixed 4 error(s) in actions/requests.ts
Fixed 4 error(s) in components/atoms/Hash.tsx
Fixed 3 error(s) in sagas/index.ts
Fixed 3 error(s) in pages/Work/WorkTabs.tsx
Fixed 3 error(s) in pages/index.ts
Fixed 3 error(s) in components/atoms/SearchInput.tsx
Fixed 2 error(s) in pages/Landing/Layout.tsx
Fixed 2 error(s) in extensions/Window.ts
Fixed 2 error(s) in reducers/index.ts
Fixed 2 error(s) in selectors/fetch.ts
Fixed 1 error(s) in bootstrap.ts
Fixed 1 error(s) in configuration.ts
Fixed 1 error(s) in components/ClassNameProps.ts
Fixed 1 error(s) in enums/FetchStatus.ts

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,7 +5,11 @@
   "scripts": {
     "start": "node devServer.js",
     "build": "webpack",
-    "serve": "ws --spa index.html -d ./dist"
+    "serve": "ws --spa index.html -d ./dist",
+    "tslint": "tslint -p ./tsconfig.json --fix",
+    "tslint:no-fix": "tslint -p ./tsconfig.json",
```
Comment (warrenv) [inline package.json:10]: Let's call this 'lint' instead of 'tslint:no-fix' to separate *what* we are doing from *how* we are doing it.

Comment (wzalazar) [reply package.json:10]: ok, it's ok for you ```tslint``` and ```lint``` ?
If we take this commands we need to create a task to review this in all projects.

Comment (warrenv) [reply package.json:10]: yes, both are fine for now. We will remove tslint in a future PR. 
Small steps :-)

Comment (wzalazar) [reply package.json:10]: ok, it's only reporting now

Comment (warrenv) [reply package.json:10]: Did you push a change?

**Review by warrenv — APPROVED** [review]

**Pull request closed without merging**